crates/kornia-3d/src/io/pcd/parser.rs

```diff
@@ -11,53 +11,164 @@ pub enum PcdError {
     #[error("Failed to read PCD file")]
     Io(#[from] std::io::Error),
 
-    /// Failed to deserialize PCD file
-    #[error("Failed to deserialize PCD file")]
-    Deserialize(#[from] bincode::error::DecodeError),
-
-    /// Unsupported PCD property
-    #[error("Unsupported PCD property")]
+    /// Unsupported or malformed PCD header
+    #[error("Unsupported PCD header")]
     UnsupportedProperty,
 
     /// Invalid PCD file extension
     #[error("Invalid PCD file extension. Got:{0}")]
     InvalidFileExtension(String),
 }
 
-/// A property of a point in a PCD file.
-#[derive(Debug, Deserialize, bincode::Decode)]
-pub struct PcdPropertyXYZRGBNCurvature {
-    /// x coordinate
-    pub x: f32,
-    /// y coordinate
-    pub y: f32,
-    /// z coordinate
-    pub z: f32,
-    /// rgb color
-    pub rgb: u32,
-    /// normal x coordinate
-    pub nx: f32,
-    /// normal y coordinate
-    pub ny: f32,
-    /// normal z coordinate
-    pub nz: f32,
-    /// curvature
-    pub curvature: f32,
+/// Describes a single field in a PCD point record
+#[derive(Debug)]
+struct PcdField {
+    name: String,
+    offset: usize, // byte offset within a point
+    size: usize,   // size of one element (bytes)
+    count: usize,  // number of elements
+    kind: char,    // 'F', 'U', or 'I'
+}
+
+#[derive(Debug)]
+struct PcdLayout {
+    fields: HashMap<String, PcdField>,
+    point_step: usize, // total bytes per point
+    num_points: usize,     // number of points
+}
+
+impl PcdLayout {
+    fn get_field_offset(&self, name: &str) -> Result<usize, PcdError> {
+        self.fields
+            .get(name)
+            .map(|f| f.offset)
+            .ok_or(PcdError::UnsupportedProperty)
+    }
+}
+
+/// Read a little-endian f32 from a byte buffer
+#[inline]
+fn read_f32(buf: &[u8], offset: usize) -> Result<f32, PcdError> {
+    let slice = buf.get(offset..offset + 4).ok_or(PcdError::UnsupportedProperty)?;
+    let mut bytes = [0u8; 4];
+    bytes.copy_from_slice(slice);
+    Ok(f32::from_le_bytes(bytes))
+}
+
+/// Read a little-endian u32 from a byte buffer
+#[inline]
+fn read_u32(buf: &[u8], offset: usize) -> Result<u32, PcdError> {
+    let slice = buf.get(offset..offset + 4).ok_or(PcdError::UnsupportedProperty)?;
+    let mut bytes = [0u8; 4];
+    bytes.copy_from_slice(slice);
+    Ok(u32::from_le_bytes(bytes))
+}
+
+
+fn parse_pcd_header<R: BufRead>(reader: &mut R) -> Result<PcdLayout, PcdError> {
```

Comment:
```suggestion
fn parse_pcd_layout<R: BufRead>(reader: &mut R) -> Result<PcdLayout, PcdError> {
```
to go according with the return struct name